src/lib.rs

```diff
@@ -2325,6 +2345,39 @@ impl Build {
         .map(|x| x.to_owned()))
     }
 
+    /// Some platforms have multiple, compatible, canonical prefixes. Look through
+    /// each possible prefix for a compiler that exists and return it. The prefixes
+    /// should be ordered from most-likely to least-likely.
+    fn find_working_gnu_prefix(&self, prefixes: &[&'static str]) -> Option<&'static str> {
+        let windows_host = self
+            .get_host()
+            .map(|host| host.contains("windows"))
+            .unwrap_or(false);
+        let suffix = if self.cpp { "-g++" } else { "-gcc" };
+        let extension = if windows_host { ".exe" } else { "" };
+
+        for prefix in prefixes {
+            let target_compiler = format!("{}{}{}", prefix, suffix, extension);
+
+            if env::var_os("PATH")
+                .as_ref()
+                .and_then(|path_entries| {
+                    env::split_paths(path_entries)
+                        .find(|path_entry| path_entry.join(&target_compiler).exists())
+                })
+                .is_some()
+            {
+                return Some(*prefix);
+            }
+        }
```

Comment:
Could the order of the loops be inverted here? That way if `/usr/bin` is near the front of `PATH` we won't have to search for everything everywhere that doesn't exist.

Reply:
I've made this change as requested.